lib/components/user/user-account-screen.js

```diff
@@ -72,21 +66,93 @@ function cloneWithHomeAndWorkAsTopLocations (loggedInUser) {
  * This screen handles creating/updating OTP user account settings.
  */
 class UserAccountScreen extends Component {
-  _updateUserPrefs = async userData => {
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      // Capture whether user is a new user at this stage, and retain that value as long as this screen is active.
+      // Reminder: When a new user progresses through the account steps,
+      // isNewUser(loggedInUser) will change to false as the database gets updated.
+      isNewUser: isNewUser(props.loggedInUser),
```

Comment:
I get why this is being done, but it just feels hacky to use component state here. Since this value never changes due to some event that gets triggered during the use of the component, it feels like this would be better handled in the application state.

Reply:
Hmm how about placing that in a class member instead? `this.isNewUser = isNewUser(...)`

Reply:
That's a little better, but still feel like it could be more straightforward for all of the state data could be handled within the application state somehow.